### PR TITLE
[Bug] java.lang.NoClassDefFoundError: org/apache/hadoop/hive/common/ValidWriteIdList in HiveDelegationTokenProvider#initialize

### LINKED ISSUE #2879: [Bug] java.lang.NoClassDefFoundError: org/apache/hadoop/hive/common/ValidWriteIdList in HiveDelegationTokenProvider#initialize
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

java.lang.NoClassDefFoundError: org/apache/hadoop/hive/common/ValidWriteIdList in HiveDelegationTokenProvider#initialize 

### Affects Version(s)

master

### Kyuubi Server Log Output

```logtalk
2022-06-14 17:44:45.861 INFO org.apache.hadoop.hive.conf.HiveConf: Found configuration file file:/private/etc/kyuubi/conf_newsbeta/hive-site.xml
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/hadoop/hive/common/ValidWriteIdList
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at org.apache.hadoop.hive.metastore.utils.JavaUtils.getClass(JavaUtils.java:52)
	at org.apache.hadoop.hive.metastore.RetryingMetaStoreClient.getProxy(RetryingMetaStoreClient.java:146)
	at org.apache.hadoop.hive.metastore.RetryingMetaStoreClient.getProxy(RetryingMetaStoreClient.java:104)
	at org.apache.kyuubi.credentials.HiveDelegationTokenProvider.initialize(HiveDelegationTokenProvider.scala:54)
	at org.apache.kyuubi.credentials.HadoopCredentialsManager.$anonfun$initialize$1(HadoopCredentialsManager.scala:104)
	at org.apache.kyuubi.credentials.HadoopCredentialsManager.$anonfun$initialize$1$adapted(HadoopCredentialsManager.scala:103)
	at scala.collection.immutable.Map$Map2.filterImpl(Map.scala:282)
	at scala.collection.immutable.Map$Map2.filterImpl(Map.scala:222)
	at scala.collection.TraversableLike.filter(TraversableLike.scala:395)
	at scala.collection.TraversableLike.filter$(TraversableLike.scala:395)
	at scala.collection.AbstractTraversable.filter(Traversable.scala:108)
	at org.apache.kyuubi.credentials.HadoopCredentialsManager.initialize(HadoopCredentialsManager.scala:103)
	at org.apache.kyuubi.service.CompositeService.$anonfun$initialize$1(CompositeService.scala:40)
	at org.apache.kyuubi.service.CompositeService.$anonfun$initialize$1$adapted(CompositeService.scala:40)
	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
	at org.apache.kyuubi.service.CompositeService.initialize(CompositeService.scala:40)
	at org.apache.kyuubi.session.SessionManager.initialize(SessionManager.scala:268)
	at org.apache.kyuubi.session.KyuubiSessionManager.initialize(KyuubiSessionManager.scala:59)
	at org.apache.kyuubi.service.CompositeService.$anonfun$initialize$1(CompositeService.scala:40)
	at org.apache.kyuubi.service.CompositeService.$anonfun$initialize$1$adapted(CompositeService.scala:40)
	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
	at org.apache.kyuubi.service.CompositeService.initialize(CompositeService.scala:40)
	at org.apache.kyuubi.service.AbstractBackendService.initialize(AbstractBackendService.scala:209)
	at org.apache.kyuubi.service.CompositeService.$anonfun$initialize$1(CompositeService.scala:40)
	at org.apache.kyuubi.service.CompositeService.$anonfun$initialize$1$adapted(CompositeService.scala:40)
	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
	at org.apache.kyuubi.service.CompositeService.initialize(CompositeService.scala:40)
	at org.apache.kyuubi.service.Serverable.initialize(Serverable.scala:46)
	at org.apache.kyuubi.server.KyuubiServer.initialize(KyuubiServer.scala:139)
	at org.apache.kyuubi.server.KyuubiServer$.startServer(KyuubiServer.scala:60)
	at org.apache.kyuubi.server.KyuubiServer$.main(KyuubiServer.scala:97)
	at org.apache.kyuubi.server.KyuubiServer.main(KyuubiServer.scala)
Caused by: java.lang.ClassNotFoundException: org.apache.hadoop.hive.common.ValidWriteIdList
	at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	... 40 more
```
```


### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

```yaml
kyuubi.authentication KERBEROS
kyuubi.kinit.principal xxx@xxx
kyuubi.kinit.keytab xxx.keytab
```

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by zhouyifan279**
### _Why are the changes needed?_
Fix #2879 

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master

**Comment by pan3793** [comment]
cc @zwangsheng as you are the original reporter.

File: pom.xml [hunk 1]
```diff
@@ -127,6 +127,7 @@
         <hive.engine.hive.version>3.1.3</hive.engine.hive.version>
         <hive.version>3.1.3</hive.version>
         <hive.service.rpc.version>3.1.3</hive.service.rpc.version>
+        <hive.storage-api.version>2.7.0</hive.storage-api.version>
```
Comment (zwangsheng) [inline pom.xml:130]: I am not sure we need storage-api for 2.7.0 version, according to *mvn dependency tree*. It seems we have 2.7.2 version jar.

**Review by zwangsheng — COMMENTED** [review]
LGTM, there are some small doubts.

File: pom.xml (hunk 1 above)
Comment (zhouyifan279) [reply pom.xml:130]: 1. hive-standalone-metastore:3.1.3 depends on hive-storage-api:2.7.0
2. hive-storage-api:2.7.2 is used in kyuubi-hive-jdbc

So, I think hive-storage-api:2.7.0 is more appropriate.

**Pull request closed without merging**